### PR TITLE
RegularPolygonMarker and CircleMarker are instances of Marker

### PR DESCRIPTION
**Pull request by BibMartin**
Adresses #259 based on @jelmelk 's great PR #265.
(BTW @jelmelk , you did not create a branch in your fork, so that I could not use your commits.)

### CODE REVIEW

File: folium/map.py [hunk 1]
```diff
@@ -363,11 +363,8 @@ def __init__(self, location, popup=None, icon=None):
         ----------
         location: tuple or list, default None
             Latitude and Longitude of Marker (Northing, Easting)
-        popup: string or tuple, default 'Pop Text'
-            Input text or visualization for object. Can pass either text,
-            or a tuple of the form (Vincent object, 'vis_path.json')
-            It is possible to adjust the width of text/HTML popups
-            using the optional keywords `popup_width` (default is 300px).
```
Comment (ocefpaf) [inline folium/map.py]: I like the slim docstring version :+1: 

However, it is not clear that people can change the `popup_width` from the Marker class.  (I am OK with that.)

File: folium/map.py [hunk 2]
```diff
@@ -378,15 +375,16 @@ def __init__(self, location, popup=None, icon=None):
         Example
         -------
         >>>map.simple_marker(location=[45.5, -122.3], popup='Portland, OR')
-        >>>map.simple_marker(location=[45.5, -122.3], popup=(vis, 'vis.json'))
-
+        >>>map.simple_marker(location=[45.5, -122.3], popup=folium.Popup('Portland, OR'))
```
Comment (ocefpaf) [inline folium/map.py]: Add a space `>>> map` so we can get this properly rendered later when building the docs.

**Comment by ocefpaf** [comment]
Thanks @jelmelk for kicking this off and @BibMartin for scoring the goal.

Before I merge this what do you think we can do about `PolyLine` and `MultiPolyLine`?  I am OK adding popups to them manually, but should we address them to keep it consistent with the `Marker`s?

**Comment by jelmelk** [comment]
> Adresses #259 based on @jelmelk 's great PR #265.
(BTW @jelmelk , you did not create a branch in your fork, so that I could not use your commits.)

It's not a problem, I'm very new to this whole thing.

File: folium/map.py (hunk 2 above)
Comment (BibMartin) [inline folium/map.py]: Done, and changed `map.simple_marker` by `Marker`.

**Comment by BibMartin** [comment]
@ocefpaf 
> Before I merge this what do you think we can do about PolyLine and MultiPolyLine?

Yes, this is more consistent. Done.

@jelmelk 
> It's not a problem, I'm very new to this whole thing.

That's great to have you in. I've discovered that a few months ago, and have a lot of fun.

**Comment by ocefpaf** [comment]
Merging this.  Thanks @jelmelk and @BibMartin!

**Pull request merged**